src/dapp/libexec/dapp/dapp-create

```diff
@@ -16,8 +16,36 @@ for i in "$@"; do
   esac
 done
 
-bin=$DAPP_OUT/${1?missing contract name}.bin
-type=$(seth --abi-constructor <"$DAPP_OUT/$1.abi")
+[[ -e "${DAPP_JSON}" ]] || {
+  echo >&2 "${0##*/}: error: ${DAPP_JSON} file not found. Did you run ``dapp build``?"
+  exit 1
+}
+
+name=${1?missing contract name}
+
+mapfile -t contracts < <(<"$DAPP_JSON" jq '.contracts|keys[]' -r)
+
+name=()
+for path in "${contracts[@]}"; do
+  [[ "${path#*:}" = "$1" ]] && name+=("$path")
+done
+
+[[ "${#name[@]}" -eq 0 ]] && {
+  echo >&2 "${0##*/}: error: $1 not found"
+  exit 1
+}
+
+[[ "${#name[@]}" -ge 2 ]] && {
+  echo >&2 "${0##*/}: error: $1 found in multiple files"
+  printf '%s\n' "${name[@]}"
+  exit 1
+}
+
+data=$(<"$DAPP_JSON" jq '.contracts' -r)
```

Comment:
We could maybe ensure that the jq selector has matched something by using the -e flag and checking for its exit status. Otherwise we're trusting that the file has the structure we expect, which might not necessarily be the case (also wrt backwards/forward-compatibility).

WDYT?

Reply:
yep, good catch! I'll add it

Reply:
Actually... `mapfile -t contracts < <(<"$DAPP_JSON" jq '.contracts|keys[]' -r)` already checks this right? And if it didn't find anything `contracts` will be length 0, and it will fail.